bleak/backends/bluezdbus/client.py

```diff
@@ -103,109 +130,284 @@ async def connect(self, **kwargs) -> bool:
+            assert_reply(reply)
+
+            interfaces_and_props: Dict[str, Dict[str, Variant]] = reply.body[0]
+
+            # The device may have been removed from BlueZ since the time we stopped scanning
+            if self._device_path not in interfaces_and_props:
+                # Sometimes devices can be removed from the BlueZ object manager
+                # before we connect to them. In this case we try using the
+                # org.bluez.Adapter1.ConnectDevice method instead. This method
+                # requires that bluetoothd is run with the --experimental flag
+                # and is available since BlueZ 5.49.
+                logger.debug(
+                    f"org.bluez.Device1 object not found, trying org.bluez.Adapter1.ConnectDevice ({self._device_path})"
+                )
+                reply = await asyncio.wait_for(
+                    self._bus.call(
+                        Message(
+                            destination=defs.BLUEZ_SERVICE,
+                            interface=defs.ADAPTER_INTERFACE,
+                            path=f"/org/bluez/{self._adapter}",
+                            member="ConnectDevice",
+                            signature="a{sv}",
+                            body=[
+                                {
+                                    "Address": Variant(
+                                        "s", self._device_info["Address"]
+                                    ),
+                                    "AddressType": Variant(
+                                        "s", self._device_info["AddressType"]
+                                    ),
+                                }
+                            ],
+                        )
+                    ),
+                    timeout,
+                )
+
+                # FIXME: how to cancel connection if timeout?
+
+                if (
+                    reply.message_type == MessageType.ERROR
+                    and reply.error_name == ErrorType.UNKNOWN_METHOD.value
+                ):
+                    logger.debug(
+                        f"org.bluez.Adapter1.ConnectDevice not found ({self._device_path}), try enabling bluetoothd --experimental"
                     )
+                    raise BleakError(
+                        "Device with address {0} could not be found. "
+                        "Try increasing `timeout` value or moving the device closer.".format(
+                            self.address
+                        )
+                    )
+
+                assert_reply(reply)
+            else:
+                # required interface
+                self._properties = unpack_variants(
+                    interfaces_and_props[self._device_path][defs.DEVICE_INTERFACE]
                 )
+
+                # optional interfaces - services and characteristics may not
+                # be populated yet
+                for path, interfaces in interfaces_and_props.items():
```

Comment:
It seems like this came up in testing. Since it seems to be working, I'm inclined to not touch it for now. If I recall, there were some subtle differences with the similar code in `get_services` that made me decide not to try to reuse the code.